dependency/dependency_sqlite.go

```diff
@@ -0,0 +1,80 @@
+package dependency
+
+import (
+	"database/sql"
+
+	"github.com/vacp2p/mvds/state"
+)
+
+// Verify that MessageDependency interface is implemented.
+var _ MessageDependency = (*sqliteDependency)(nil)
+
+type sqliteDependency struct {
+	db *sql.DB
+}
+
+func NewPersistentDependency(db *sql.DB) *sqliteDependency {
+	return &sqliteDependency{db: db}
+}
+
+func (sd *sqliteDependency) Add(msg, dependency state.MessageID) error {
+	_, err := sd.db.Exec(`INSERT INTO mvds_dependencies (msg, dependency) VALUES (?, ?)`, msg[:], dependency[:])
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (sd *sqliteDependency) Dependants(id state.MessageID) ([]state.MessageID, error) {
+	rows, err := sd.db.Query(`SELECT msg FROM mvds_dependencies WHERE dependency = ?`, id[:])
+	if err != nil {
+		return nil, err
+	}
+
+	defer rows.Close()
+
+	msgs := make([]state.MessageID, 0)
```

Comment:
```suggestion
	var msgs []state.MessageID
```

the same result but no memory allocated if no data in DB.